### PR TITLE
Concat data update - addresses issue #262

### PR DESCRIPTION
**Pull request by rstoneback**
# Description

Addresses # 262

Removes deprecated argument from pandas' concat method. 
Added a test to ensure method is regularly run.

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)

# How Has This Been Tested?

I encountered the bug using the multi_file_day option while processing COSMIC-2 data. Pandas informed me that sort=True no longer supported. 

**Test Configuration**:
* Operating system
* Version number pandas 0.20.
* Any details about your local setup that are relevant
 I was below setup.py reqs. Apparently python setup.py develop doesn't really enforce requirements. Issue that cropped up due to a development environment with heritage.

# Checklist:

- [X] Make sure you are merging into the ``develop`` (not ``master``) branch
- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation (No changes needed)
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes
- [X] Any dependent changes have been merged and published in downstream modules
- [X] Add a note to ``CHANGELOG.md``, summarizing the changes

### CODE REVIEW

File: pysat/_instrument.py [hunk 1]
```diff
@@ -701,7 +701,7 @@ def concat_data(self, data, *args, **kwargs):
         """Concats data1 and data2 for xarray or pandas as needed"""
 
         if self.pandas_format:
-            return pds.concat(data, sort=True, *args, **kwargs)
+            return pds.concat(data, *args, **kwargs)
```
Comment (rstoneback) [inline pysat/_instrument.py]: This is the relevant change.

File: pysat/tests/test_instrument.py [hunk 2]
```diff
@@ -174,6 +174,30 @@ def test_filtered_date_attribute(self):
         self.testInst.date = now
         assert today == self.testInst.date
 
+
```
Comment (rstoneback) [inline pysat/tests/test_instrument.py]: Relevant changes below.

**Comment by rstoneback** [comment]
This pull incorrectly includes code from pull #261

File: CHANGELOG.md [hunk 3]
```diff
@@ -2,14 +2,18 @@
...
 
-## [2.X.X] - 2019-07-17
+## [2.X.X] - 2019-08-20
 - New Features
    - Added new velocity format options to utils.coords.scale_units
    - Improved failure messages for utils.coords.scale_units
    - Added some tests for model_utils
+   - Added option to to_netCDF that names variables in the written file
+   based upon the strings in the Instrument.meta object
+   - Improved compatibbility with NASA ICON's file standards
```
Comment (rstoneback) [inline CHANGELOG.md]: Typo fixed in other pull.

**Comment by jklenzing** [comment]
See comments in #262 before merging.  `sort` is a new feature, rather than a deprecated one.

**Comment by rstoneback** [comment]
I fixed the code back up, restoring sort=True. Now this pull only adds a test for the concat_data method.

**Comment by rstoneback** [comment]
Thanks for reminding me of the code history.

After a bit more time with the code I think we should go back to leaving sort=True out. The concat_data Instrument method is otherwise set up to pass extra args and keyword args through to the underlying concat methods. Fixing one argument seems inconsistent.

I do appreciate that sort=True was put in to maintain old behaviors. That being said, the places in pysat where it is used doesn't rely on sort=True. The test I wrote, as part of this pull, no longer works since the columns are in a different order than the original data. I could change up the tests, but now we have different behaviors for pandas and xarray.

**Comment by jklenzing** [comment]
I'm not opposed to pulling this out.  We just need to make sure that the code does what we expect.  As of pandas 0.25.0: "The current default of sorting is deprecated and will change to not-sorting in a future version of pandas."

An advantage to leaving it out is that xarray does not (currently) have this as a keyword.

http://xarray.pydata.org/en/stable/generated/xarray.concat.html

**Comment by rstoneback** [comment]
I was thinking along the same lines. It's good to have some consistency across our data formats.

I will poke at the behavior in a few more situations and report back.

File: pysat/tests/test_instrument.py (hunk 2 above)
Comment (rstoneback) [reply pysat/tests/test_instrument.py]: This comment no longer needed now that the ICON updates have been properly merged. All code differences are authentic now.

**Comment by rstoneback** [comment]
This is ready for review. Added @jklenzing as he already chimed in (Thanks!) due to recent work in this area. The default behavior is the same across datasets, pandas and xarray. For pandas, this requires setting sort to False, and for xarray specifying the time dimension. Support was also added so users can pass in their own values for these keyword arguments.

Since this pull was long, I can summarize. I ran into an issue caused by an out of date version of pandas. The pull was then redirected to improve consistency and add unit tests to the concat_data_method instead. A data access issue was encountered with xarray in the unit tests and was fixed by @asher-pembroke.

File: CHANGELOG.md [hunk 4]
```diff
@@ -19,7 +19,9 @@ This project adheres to [Semantic Versioning](http://semver.org/).
 - Bug fix
    - Fixed implementation of utils routines in model_utils and jro_isr
    - Fixed error catching bug in model_utils
+   - Removed deprecated keyword from methods within Instrument.concat_data
```
Comment (jklenzing) [inline CHANGELOG.md]: doesn't describe latest changes

File: pysat/tests/test_instrument.py [hunk 5]
```diff
@@ -174,6 +174,59 @@ def test_filtered_date_attribute(self):
...
+        # data set #1
+        data1 = self.testInst.data
+        len1 = len(self.testInst.index)
+        # concat together
+        self.testInst.data = self.testInst.concat_data([data1, data2])
+        # test for concatenation
+        len3 = len(self.testInst.index)
+        assert (len3 == len1 + len2)
+
+        # concat together with sort=True
```
Comment (jklenzing) [inline pysat/tests/test_instrument.py]: is this comment for both `if` and `else`?

**Pull request merged**